src/components/pages/SignupPage.js

```diff
@@ -0,0 +1,180 @@
+import React, { useEffect, useState } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+// import { NavLink } from 'react-router-dom';
+import cx from 'classnames';
+import colorScheme from '../../colorScheme';
+import style from './signup.module.css';
+import hitAPIWithSignupDetails from '../../redux/user/user';
+
+const SignupPage = () => {
+  console.log(process.env, 'process.env');
+  const {
+    formHeader,
+    form,
+    formGroup,
+    formControl,
+    label,
+    btn,
+    h2,
+    signedUpMessage,
+    noSignedUpMessage,
+    // p,
+    error,
+    orGroup,
+    line,
+    or,
+  } = style;
+
+  const dispatch = useDispatch();
+  const state = useSelector((state) => state.user);
+  console.log(state, 'state form signup');
+  const { signedUp } = state;
+  const [input, setInput] = useState({
+    name: '',
+    email: '',
+    password: '',
+    confirmPwd: '',
+  });
+
+  const [missmatch, setMissmatch] = useState('');
+  const [signUpSuccess, setSignUpSucess] = useState(signedUp);
+
+  const handleInput = (event) => {
+    const { name, value } = event.target;
+    if (name === 'password' || name === 'confirmPwd') setMissmatch(() => '');
+    setInput((prevInput) => ({
+      ...prevInput,
+      [name]: value,
+    }));
+  };
+
+  const handleSubmit = (event) => {
+    setSignUpSucess(() => false);
+    const {
+      name,
+      email,
+      password,
+      confirmPwd,
+    } = input;
+    if (name && email && password && confirmPwd) {
+      event.preventDefault();
+      if (password !== confirmPwd) {
+        return setMissmatch(() => 'Password mismatch!');
+      }
+      if (password.length < 6) {
+        return setMissmatch(() => 'Password too short!');
+      }
+      dispatch(hitAPIWithSignupDetails(input));
+    }
+    return true;
+  };
+
+  useEffect(() => {
+    setSignUpSucess(() => signedUp);
+  }, [state]);
+
+  return (
+    <form className={cx(colorScheme.green, form)}>
+      <div style={{ backgroundColor: colorScheme.blue }} className={cx(formHeader)}>
+        <h2 className={h2}>Let&apos;s Sign Up</h2>
+        {/* <p className={p}>Fill out this form to sign up with us!</p> */}
```

Comment:
- [ ]  [OPTIONAL] Kindly remove unused lines of code in order to have a clean code-base.